Improve contrib guide (#311)
kmyk/online-judge-tools: CONTRIBUTING.md

## issues / issue について

Not only sending Pull Requests, feature requests and bug reports are welcome.

機能要求やバグ報告は気軽にしてください。
コードを書くことだけが開発ではありません。

## pull requests / プルリクについて

PR is always welcome.

However, please note that PR is not always merged as it is.
To improve PR quality, reviewers may ask you change requests.

-   Test your PR branch on local by `python3 setup.py test`.

> -   Write friendly code to the reviewers.

Author: developersの方がいいかもしれません。

Reviewer: ```suggestion
-   Write code easy to understand.
```

「やさしい」の訳の "friendly" だけど、違う意味になりそうな気がするので意訳したい

>     -   Diffs need to be limited to the PR purpose.   

Reviewer: ```suggestion
    -   Don't make diff which is unnecessary for the purpose of PR.
```

列挙されてる中でこれだけ命令文でないので型を揃えたい。書きにくかったので大きめの意訳

    -   Split commits appropriately.
    -   Comment on the code where you are not confident of.

> -   If you want to add feature, it is desirable discuss in advance that before writing code.

Author: gitterのリンクを貼ったほうがいいですかね。

Reviewer: (あってだめではないですが、) なくていいと思います。
このファイルを見る人なら容易にリンクを見つけられるだろうし、もしこれを貼るなら他のファイル名などにもリンクを貼りたくなってきて面倒だからです。

> -   If you want to add feature, it is desirable discuss in advance that before writing code.

Reviewer: ```suggestion
-   If you want to add a feature, it would be better to discuss before writing code
    -   because your feature is not always merged
```

-  `feature` は可算ぽい https://eow.alc.co.jp/search?q=feature
-   `in advance` と `before` は被ってる

## issues / issue について

Not only sending Pull Requests, feature requests and bug reports are welcome.

機能要求やバグ報告は気軽にしてください。
コードを書くことだけが開発ではありません。

## pull requests / プルリクについて

> PR is always welcome.

Reviewer: ```suggestion
```

ところでこれ消した方がいいかもしれない。
「プルリクは歓迎します」というのはそのようなことを滅多にしてこない日本人に対しての最速という性格があるので、その意味では英語に翻訳した場合は消えるべきです。

However, please note that PR is not always merged as it is.
To improve PR quality, reviewers may ask you change requests.

-   Test your PR branch on local by `python3 setup.py test`.
-   Write code easy to understand.
    -   Don't make diff which is unnecessary for the purpose of PR.
    -   Split commits appropriately.

>     -   Comment on the code where you are not confident of.

Reviewer: ここの `where` って適切ですか？

Author: https://help.duda.co/hc/en-us/articles/115000553108-I-have-custom-code-where-do-I-place-it-

にという文章があります。
When getting the code from another source, it's important to understand where that code must be placed.

Reviewer: その例文は根拠としては不適切に見えます。なぜなら `where` の型が違います。
`where that code must be placed` の `where` は全体で名詞となる部分を作るために働いていますが、元々話題にしていた文の `where` はそうではありません。

Author: https://www.cs.utah.edu/~germain/PPS/Topics/commenting.html

`Any "tricky" code where it is not immediately obvious what you are trying to accomplish, should have comments right above it or on the same line with it.`
このwhereはどうでしょう。なお、型とか忘れてます。

## philosophy of design / 設計の方針

Our first purpose is "to help get the top in the programming contests".
To make it possible, online-judge-tools automates "mistakable and cumbersome manual work".
Also, we are focusing on "online-judge-tools does not cause the penalties".

> 第一義は「コンテストで上位を取ることに役立つこと」です。

Author: ここの並びは変だと思いました。

Reviewer: わかる

## philosophy of design / 設計の方針

> Our first purpose is "to help get the top in the programming contests".

Reviewer: ```suggestion
Our purpose is "to help get a higher rank in programming contests".
```

-   英語の `first` に日本語の「第一」みたいな意味はあるのかな？
   どうなのかは知らないのだけど、 "second purpose" が期待されそうに感じるしとりあえず消しておきたい
-  `the top` って言うと「1位」しか含まなさそうな気がする。たぶん

Author: google翻訳先生が上位を目指すをaim for topと出したんですが、上位という意味はないですね。

To make it possible, online-judge-tools automates "mistakable and cumbersome manual work".

Also, we are focusing on "online-judge-tools does not cause the penalties".

> Since online-judge-tools using web scraping, sometimes behavior is unstable.

Reviewer: ```suggestion
Since online-judge-tools does web scraping, the behavior is unstable.
```

-   `sometimes` の位置が変
-   そもそもこの場合の「不安定」は「個々の実行」の性質ではなくどちらかと言えば「プログラムそのもの」の性質のはずで「ときどき不安定」というのは奇妙に聞こえる

> To prevent penalties under this condition, "we avoid to supports the features which is prone to malfunction" and "make it easy to notice malfunction if malfunction is occured". 

Reviewer: これなんだか怪しげ
日本語だと括弧はわりと雑に使うけど、英語のクォーテーションはこのような「強調」とか「文の構造を明示する目的」には使われない気がする。たぶん

Author: ですね。マークダウンの強調を使うことにしました。

> For example, the sample cases which is got by online-judge-tools is done pretty printing.

Reviewer: ```suggestion
For example, this program pretty-prints the downloaded sample cases.
```

`the sample cases which is ... is ...` は不必要に分かりにくい

## philosophy of design / 設計の方針

**Our purpose is to help get a higher rank in programming contests.**

> To make it possible, online-judge-tools automates *mistakable and cumbersome manual work*.

Reviewer: このあたり `online-judge-tools` とそのまま書いてるけど、 `this program` とかの方が自然だったりしないかな？ (実際にどうなのかは私も知りません)

Author: https://github.com/Microsoft/LightGBM
読むとLightGBMということばが何回か出てきます。

Reviewer: たしかに出てきてはいますが、それは根拠として弱いように感じます。
readmeの一番上の部分で書かれる文章はすこし性質が違うためです。また `LightGBM v2` `LightGBM Slack team` などという形での出現は固有名詞であるため数え入れるべきではありません。

Reviewer: でも一応はあるということだしこのままでもいいかも？

Author: https://github.com/tensorflow/tensorflow

tensorflowと何度も書いてます。

Also, we are focusing on *online-judge-tools does not cause the penalties*.
Since online-judge-tools does web scraping, the behavior is unstable.

> To prevent penalties under this condition, *we avoid to supports the features which is prone to malfunction* and *make it easy to notice malfunction if malfunction is occured*. 

Reviewer: -   `condition` って言うと「条件」とか「状態」という雰囲気がつよくて適当ではなさそう？ https://ejje.weblio.jp/content/condition
-   `*we avoid to supports the features` 後ろの強調が `*make it easy ...` なのでこっちも `*avoid to ...` にしたい。 s/supports/support/
-   `the features which is prone to malfunction` もうちょっと平たく書いてほしいです。 s/is/are/
-   `notice malfunction if malfunction is occured` これ `malfuction` を2回書いてるのやめたい

For example, this program pretty-prints the downloaded sample cases.

> If such a printing does not exist, false acquisition of the sample cases causes a considerable time loss.

Reviewer: `false acquisition` なんかもうちょっと平易に書きたい

Author: `Sample case acquisition error`はどうでしょう？

Reviewer: もう一声ほしいです。

元の日本語が「サンプルケースの取得ミス」ですが、`false acquisition of the sample cases` だと 「サンプルケースの偽獲得」ぐらいの雰囲気でした。今回の `Sample case acquisition error` でもまだ「サンプルケース-獲得エラー」ぐらいの固い感じが残っているように思います。

Author: 代案思いつかないので、このままにするか、一旦ここは英訳せずに消します。

## CI

> Travis CI will run automatically if you commit or send PR on `master` or `develop` branch.

Reviewer: ```suggestion
Travis CI will run automatically when you commit or send PR on `master` or `develop` branch.
```

伝わるけど

> Since the same test is done by `python3 setup.py test`, if you tested locally, you don't have to mind it.

Reviewer: この文かなり雑だし改善したい
(元の日本語の文章もまあ雑なので無視して書き直すのがいい気がします)

Author: 気にしなくていい、というのも変な気がしたので、とりあえず同等のテストをしますという記述にとどめました。

Reviewer: :+1:

## CI

Travis CI will run automatically when you commit or send PR on `master` or `develop` branch.

> The same test as that by `python3 setup.py test` is executed.

Reviewer: すっきりして良くなったように思います。
しかしまだもう一歩で、 "as that by ..." はなにか変です。

例えば `the same thing as that by` などで検索するとまったく使われていない表現であることが分かります https://www.google.com/search?q=%22the+same+thing+as+that+by%22

`master` `develop` に関する commit や pull request について CI が走ります。

> `python3 setup.py test` の実行によるものと同等のテストが行われます。

Reviewer: これいい感じ
